include/swift/SIL/SILArgument.h

```diff
@@ -36,7 +36,11 @@ SILFunctionType::getSILArgumentConvention(unsigned index) const {
 
 class SILArgument : public ValueBase {
   void operator=(const SILArgument &) = delete;
+
+  // Work around MSVC error: attempting to reference a deleted function.
+#if !defined(_MSC_VER) || defined(__clang__)
   void operator delete(void *Ptr, size_t) = delete;
+#endif
```

Comment:
It might be nice to consolidate these into a compatibility macro in a `swift/Basic` header:

```
#if !defined(_MSC_VER) || defined(__clang__)
# define SWIFT_DELETE_OPERATOR_DELETE(...) \
  void operator delete(__VA_ARGS__) = delete;
#else
// Due to an MSVC bug, deleting an operator delete
// also causes the destructor to be deleted.
# define SWIFT_DELETE_OPERATOR_DELETE(...)  
#endif
```